### ISSUE TITLE
Adding album of non-monitored artist

### ISSUE THREAD

**Issue by supersnellehenk** [tidusjar/Ombi#2915]
**Describe the bug**
Error when adding an album of an artist which isn't monitored by Lidarr.

**To Reproduce**
Add an album of an artist which isn't monitored.

**Expected behavior**
Lidarr adding the album, which then monitors the artist.

**Logs (Logs directory where Ombi is located)**
```2019-04-06 21:34:48.214 +02:00 [Error] Something bad happened, ErrorMiddleware caught this
System.ArgumentNullException: Value cannot be null.
Parameter name: entity
   at Microsoft.EntityFrameworkCore.Utilities.Check.NotNull[T](T value, String parameterName)
   at Microsoft.EntityFrameworkCore.DbContext.Remove[TEntity](TEntity entity)
   at Ombi.Store.Repository.BaseRepository`2.Delete(T request) in C:\projects\requestplex\src\Ombi.Store\Repository\BaseRepository.cs:line 57
   at Ombi.Core.Engine.MovieRequestEngine.RemoveMovieRequest(Int32 requestId) in C:\projects\requestplex\src\Ombi.Core\Engine\MovieRequestEngine.cs:line 420
   at Ombi.Controllers.RequestController.DeleteRequest(Int32 requestId) in C:\projects\requestplex\src\Ombi\Controllers\RequestController.cs:line 122
   at lambda_method(Closure , Object )
   at Microsoft.AspNetCore.Mvc.Internal.ActionMethodExecutor.AwaitableResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.InvokeActionMethodAsync()
   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.InvokeNextActionFilterAsync()
   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.Rethrow(ActionExecutedContext context)
   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.InvokeInnerFilterAsync()
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeNextResourceFilter()
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.Rethrow(ResourceExecutedContext context)
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeFilterPipelineAsync()
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeAsync()
   at Microsoft.AspNetCore.Builder.RouterMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.StaticFiles.StaticFileMiddleware.Invoke(HttpContext context)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIIndexMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.Cors.Infrastructure.CorsMiddleware.InvokeCore(HttpContext context)
   at Ombi.ApiKeyMiddlewear.Invoke(HttpContext context) in C:\projects\requestplex\src\Ombi\ApiKeyMiddlewear.cs:line 51
   at Ombi.ErrorHandlingMiddleware.Invoke(HttpContext context) in C:\projects\requestplex\src\Ombi\ErrorHandlingMiddlewear.cs:line 24
2019-04-06 23:21:15.922 +02:00 [Error] Exception thrown when sending a music to DVR app, added to the request queue
System.NullReferenceException: Object reference not set to an instance of an object.
   at Ombi.Core.Senders.MusicSender.SendToLidarr(AlbumRequest model, LidarrSettings settings) in C:\projects\requestplex\src\Ombi.Core\Senders\MusicSender.cs:line 91
   at Ombi.Core.Senders.MusicSender.Send(AlbumRequest model) in C:\projects\requestplex\src\Ombi.Core\Senders\MusicSender.cs:line 42
2019-04-06 23:21:16.705 +02:00 [Warning] Tried auto sending album but failed. Message: "Something went wrong!"
2019-04-06 23:21:16.737 +02:00 [Warning] Tried auto sending Album but failed. Message: "Something went wrong!"
```

**Desktop (please complete the following information):**
 - OS: Windows 10 build 17763

**Ombi Version (please complete the following information):**
 - Version: 3.0.4256
- Media Server: Plex

**Lidarr Version**
- Version: 0.5.0.794 (Nightly)

**Comment by tidusjar** [comment]
Can you please try the dev build

**Comment by supersnellehenk** [comment]
Running 3.0.4378 now, getting a different error.

```2019-04-07 00:53:09.960 +02:00 [Error] Exception thrown when sending a music to DVR app, added to the request queue
System.NullReferenceException: Object reference not set to an instance of an object.
   at Ombi.Core.Senders.MusicSender.SendToLidarr(AlbumRequest model, LidarrSettings settings) in C:\projects\requestplex\src\Ombi.Core\Senders\MusicSender.cs:line 91
   at Ombi.Core.Senders.MusicSender.Send(AlbumRequest model) in C:\projects\requestplex\src\Ombi.Core\Senders\MusicSender.cs:line 42
2019-04-07 00:53:10.331 +02:00 [Warning] Tried auto sending album but failed. Message: "Something went wrong!"
2019-04-07 00:53:10.331 +02:00 [Warning] Tried auto sending Album but failed. Message: "Something went wrong!"
```

**Comment by tidusjar** [comment]
Can you please try build #4380 and post the results?

**Comment by supersnellehenk** [comment]
Ran build #4380, got this as a response.
```fail: Ombi.Core.Senders.MusicSender[0]
      Exception thrown when sending a music to DVR app, added to the request queue
System.ArgumentException: The string can't be null or empty.
Parameter name: ForeignArtistId
   at EnsureThat.EnsureArg.IsNotNullOrEmpty(String value, String paramName)
   at Ombi.Core.Senders.MusicSender.SendToLidarr(AlbumRequest model, LidarrSettings settings) in C:\projects\requestplex\src\Ombi.Core\Senders\MusicSender.cs:line 92
   at Ombi.Core.Senders.MusicSender.Send(AlbumRequest model) in C:\projects\requestplex\src\Ombi.Core\Senders\MusicSender.cs:line 43
warn: Ombi.Core.Engine.MusicRequestEngine[0]
      Tried auto sending album but failed. Message: Something went wrong!
warn: Ombi.Core.Engine.MusicRequestEngine[0]
      Tried auto sending Album but failed. Message: Something went wrong!```

**Comment by tidusjar** [comment]
That's strange. Have you tried different albums by different artists?

Can you please give a step by step of how you are getting to this stage?

**Comment by supersnellehenk** [comment]
I've tried
- Time 2 Shine - Da Tweekaz
- Resist - Within Temptation

which both aren't monitored in Lidarr.

I simply open up Search, go to Music, type in the album name, click request which results in a failure. Comes back each time to the ForeignArtistId.

Made a video of what I'm doing: https://www.youtube.com/watch?v=zFIAF3mhofE

**Comment by tidusjar** [comment]
This bug is due to https://github.com/lidarr/Lidarr/issues/733

**Comment by supersnellehenk** [comment]
lidarr/Lidarr#733 has been merged, solved in Lidarr Nightly version 0.5.0.810.

**Comment by tidusjar** [comment]
Excellent thanks!
